test/ProtocolFeesImplementation.t.sol

```diff
@@ -218,4 +219,30 @@ contract ProtocolFeesTest is Test, GasSnapshot, Deployers {
         assertFalse(success);
         assertEq(protocolFee, 0);
     }
+
+    function test_fetchProtocolFee_gasLimit() public {
+        gasLimitFeeController = new GasLimitProtocolFeeControllerTest();
+        protocolFees.setProtocolFeeController(gasLimitFeeController);
+        vm.prank(address(gasLimitFeeController));
+        (bool success, uint24 protocolFee) = protocolFees.fetchProtocolFee(key);
+        assertFalse(success);
+        assertEq(protocolFee, 0);
+    }
+
+    function test_fetchProtocolFee_revertsWithProtocolFeeCannotBeFetched() public {
+        protocolFees = new ProtocolFeesImplementation(9079256829993496519);
+        protocolFees.setProtocolFeeController(feeController);
+        vm.prank(address(feeController));
+        vm.expectRevert(IProtocolFees.ProtocolFeeCannotBeFetched.selector);
+        // consumes too much gas before fetching the fee
+        protocolFees.consumeGasLimitAndFetchFee(key);
+    }
+
+    function test_fetchProtocolFee_revertsWithGasLimitExceeded() public {
+        protocolFees = new ProtocolFeesImplementation(9079256829993496519);
+        protocolFees.setProtocolFeeController(feeController);
+        vm.prank(address(feeController));
+        vm.expectRevert();
```

Comment:
please can you fill in what the revert message should be for this case

Reply:
I cant see why this one will be any different to the previous one? surely it also reverts with `ProtocolFeeCannotBeFetched` because its consuming even more gas and the amount left is even lower?

Reply:
it consumes less!

Reply:
the limit is the same, but in `consumeGasAndFetchFee`, its consuming less gas

Reply:
would agree here - not sure what the difference is if there is one besides just consuming a different amount of gas before calling fetchProtocolFee

You could fuzz that amount even

Reply:
oh yeah i messed up the numbers